sourcegraph/sourcegraph
# Code Insights: Remove code insight landing page flag

Closes https://github.com/sourcegraph/sourcegraph/issues/30990

## Background

Prior to this PR, you had to use a special feature flag to see the landing page. Now it works by default without flags. 

## Tech notes
- This PR removes the legacy dashboard page component
- Tests for the dashboard page component have been moved and are now live in the `CodeInsightsRootPage.tsx`

## Test plan

- You should see in product landing page without any feature flags in your settings
- Check the standard routes for correct route redirections
- Check that tabs works correctly (you shouldn't lose selected dashboard while switching between tabs, dashboard and 
get started page tabs)
- Check that wrong URL paths like /insights or insights/dashboard are redirected on the right routes (/insights -> /insights/dashboard/all, insights/dashboard  -> /insights/dashboard/all)
